File: CPP/Program/CRUD.cpp
/*Saya Fikry Idham Dwiyana NIM 2101294 mengerjakan Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//library yang digunakan
#include <bits/stdc++.h>
#include "DPR.cpp"

using namespace std;

//deklarasi class CRUD
class CRUD
{
	public:
		//Tampungan dan penanda
		int id;
		string nama;
		string bidang;
		string partai;
		int j;

		//constructor (dibuat kosong karena tidak diperlukan untuk diisi)
		CRUD()
		{

		}

		//method untuk menambahkan data (create)
		void add(list<DPR> *llist)
		{
			//instance object menggunakan class DPR
			DPR temp;
			j = 0;

			//inputan untuk data DPR
			cout << "Masukkan Id:";
			cin >> id;
			cout << "Masukkan Nama:";
			cin >> nama;
			cout << "Masukkan Bidang:";
			cin >> bidang;
			cout << "Masukkan Partai:";
			cin >> partai;

			//perulangan sebanyak data dalam list
			for(list<DPR>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				//pengecekan jika Id yang diinput sama dengan Id pada list
				//Id digunakan sebagai primary key
				if(id == it->getId())
				{
					//print warning bahwa data sudah tersedia
					//variabel j digunakan sebagai penanda
					j = 1;
					cout << "\nTidak dapat menambahkan data, ID sudah terdaftar!" << '\n';
				}
			}

			//jika variabel j tetap nol (artinya tidak ada Id yang sama / data belum tersedia)
			if(j == 0)
			{
				//set data inputan ke object tampungan yang sebelumnya telah dibuat
				temp.setId(id);
				temp.setName(nama);
				temp.setBidang(bidang);
				temp.setPartai(partai);

				//push object tersebut ke dalam list
				(*llist).push_back(temp);
				cout << "\nData berhasil ditambahkan!" << '\n';
			}
		}

		//method untuk mengubah data (update)
		void update(int id, list<DPR> *llist)
		{
			//variabel penanda
			j = 0;

			//perulangan sebanyak data pada list
			for(list<DPR>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				//jika Id pada parameter (yaitu Id yang akan diubah) sama dengan Id pada list
				if(id == it->getId())
				{
					//inputan untuk data yang akan diubah
					cout << "Masukkan Id:";
					cin >> id;
					cout << "Masukkan Nama:";
					cin >> nama;
					cout << "Masukkan Program Bidang:";
					cin >> bidang;
					cout << "Masukkan Partai:";
					cin >> partai;

					//set inputan
					it->setId(id);
					it->setName(nama);
					it->setBidang(bidang);
					it->setPartai(partai);

					cout << "\nData berhasil diubah!" << '\n';

					j = 1;
				}
			}

			//jika variabel j tetap nol (artinya data yang akan diupdate tidak tersedia)
			if(j == 0)
			{
				//print warning bahwa data tidak tersedia
				cout << "\nData tidak tersedia!" << '\n';
			}
		}

		//method untuk menghapus data (delete)
		void del(int id, list<DPR> *llist)
		{
			j = 0;

			//perulangan sebanyak data pada list
			for(list<DPR>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				//jika Id dari parameter sama dengan Id pada list
				if(id == it->getId())
				{
					//hapus data pada list
					it = (*llist).erase(it);
					it--;
					cout << "\nData berhasil dihapus!" << '\n';
					j = 1;
				}
			}

			//jika data tidak tersedia
			if(j == 0)
			{
				cout << "\nData tidak tersedia!" << '\n';
			}
		}

		//method untuk menampilkan data (read)
		void show(list<DPR> *llist)
		{
			//output dalam bentuk tabel
			cout << "\nList Anggota:\n";
			int i = 0;
			cout << "+=======================================+\n" << "|  Id  |  Nama  |  Bidang  |  Partai  |\n" << "+=======================================+\n";
			for(list<DPR>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				cout << "| " <<(i + 1) << ". " << it->getId() << " | " << it->getName() << " | " << it->getBidang() << " | " << it->getPartai() << " |\n";
				cout << "+---------------------------------------+\n";
				i++;
			}
		}

		//destructor
		~CRUD()
		{

		}
};

File: Python/Program/CRUD.py
# Saya Fikry Idham Dwiyana NIM 2101294 mengerjakan Latihan 1
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.

#import class
from DPR import DPR

#deklarasi class CRUD
class CRUD:
	#atribut untuk tampungan
	id = ""
	nama = ""
	bidang = ""
	partai = ""

	#constructor
	def __init__():
		i = 0 #gaboleh kosong

	#method untuk menambahkan data (create)
	def add(dprri):
		j = 0

		#inputan untuk data DPR
		print("Masukkan Id:")
		id = str(input())
		print("Masukkan Nama:")
		nama = str(input())
		print("Masukkan Bidang:")
		bidang = str(input())
		print("Masukkan Partai:")
		partai = str(input())

		#pengecekan apakah data sudah tersedia pada list
		for dpr in dprri:
			if(id == dpr.getId()):
				j = 1
				print("\nTidak dapat menambahkan data, Id sudah terisi!")

		#jika tidak tersedia maha masukkan ke list
		if(j == 0):
			dprri.append(DPR(id, nama, bidang, partai))
			print("\nData berhasil ditambahkan!")

	#method untuk mengubah data (update)
	def update(id, dprri):
		j = 0

		#perulangan untuk mencari data yang akan diubah 
		for dpr in dprri:
			if(id == dpr.getId()):
				#inputan untuk data yang akan diubah
				print("Masukkan Id:")
				id = str(input())
				print("Masukkan Nama:")
				nama = str(input())
				print("Masukkan Bidang:")
				bidang = str(input())
				print("Masukkan Partai:")
				partai = str(input())

				dpr.setId(id);
				dpr.setName(nama);
				dpr.setBidang(bidang);
				dpr.setPartai(partai);

				print("\nData berhasil diubah!")

				j = 1

		#jika data tidak tersedia
		if(j == 0):
			print("\nData tidak tersedia!")

	#method untuk menghapus data (delete)
	def delete(id, dprri):
		i = 0
		j = 0

		#perulangan untuk mencari data yang akan dihapus 
		for dpr in dprri:
			if(id == dpr.getId()):
				del dprri[i]
				print("\nData berhasil dihapus!")
				j = 1
			i += 1

		#jika data tidak tersedia
		if(j == 0):
			print("\nData tidak tersedia!")

	#method untuk menampilkan data (read)
	def show(dprri):
		#output berupa tabel
		i = 0
		print("\nList DPR:")
		print("+=======================================+\n" + "|  Id  |  Nama  |  Bidang  |  Partai  |\n" + "+=======================================+")
		for dpr in dprri:
			print("| " + str(i + 1) + ". " + dpr.getId() + " | " + dpr.getName() + " | " + dpr.getBidang() + " | " + dpr.getPartai() + " |")
			print("+---------------------------------------+")
			i += 1